### ISSUE TITLE
copy-paste shortcuts don't work in explorer anymore

### ISSUE THREAD

**Issue by jrieken** [microsoft/vscode#175751]
* open explorer
* select a file
* press `cmd+c` and `cmd+v` to copy paste a file in place
* :bug: nothing happens, in the past I would get a copy of the file

**Comment by lramos15** [comment]
Hmmm. I cannot repro this and am able to copy and paste.

Do you have any custom keybindings? What is your `explorer.incrementalNaming` setting?

**Comment by jrieken** [comment]
It's the default, `simple`


this is from the keybindings log

```
2023-03-01 15:36:13.500 [info] [KeybindingService]: | Converted keydown event - modifiers: [meta], code: MetaLeft, keyCode: 57 ('Meta')
2023-03-01 15:36:13.500 [info] [KeybindingService]: \ Keyboard event cannot be dispatched in keydown phase.
2023-03-01 15:36:13.735 [info] [KeybindingService]: / Soft dispatching keyboard event
2023-03-01 15:36:13.736 [info] [KeybindingService]: | Resolving meta+[KeyC]
2023-03-01 15:36:13.736 [info] [KeybindingService]: \ From 19 keybinding entries, matched editor.action.clipboardCopyAction, when: no when condition, source: built-in.
2023-03-01 15:36:13.736 [info] [KeybindingService]: / Received  keydown event - modifiers: [meta], code: KeyC, keyCode: 67, key: c
2023-03-01 15:36:13.736 [info] [KeybindingService]: | Converted keydown event - modifiers: [meta], code: KeyC, keyCode: 33 ('C')
2023-03-01 15:36:13.736 [info] [KeybindingService]: | Resolving meta+[KeyC]
2023-03-01 15:36:13.738 [info] [KeybindingService]: \ From 19 keybinding entries, matched editor.action.clipboardCopyAction, when: no when condition, source: built-in.
2023-03-01 15:36:13.738 [info] [KeybindingService]: + Invoking command editor.action.clipboardCopyAction.
2023-03-01 15:36:13.867 [info] [KeybindingService]: + Storing single modifier for possible chord meta.
2023-03-01 15:36:13.941 [info] [KeybindingService]: / Received  keydown event - modifiers: [meta], code: MetaLeft, keyCode: 91, key: Meta
2023-03-01 15:36:13.942 [info] [KeybindingService]: | Converted keydown event - modifiers: [meta], code: MetaLeft, keyCode: 57 ('Meta')
2023-03-01 15:36:13.942 [info] [KeybindingService]: \ Keyboard event cannot be dispatched in keydown phase.
2023-03-01 15:36:14.100 [info] [KeybindingService]: / Soft dispatching keyboard event
2023-03-01 15:36:14.100 [info] [KeybindingService]: | Resolving meta+[KeyV]
2023-03-01 15:36:14.100 [info] [KeybindingService]: \ From 4 keybinding entries, matched editor.action.clipboardPasteAction, when: no when condition, source: built-in.
2023-03-01 15:36:14.100 [info] [KeybindingService]: / Received  keydown event - modifiers: [meta], code: KeyV, keyCode: 86, key: v
2023-03-01 15:36:14.100 [info] [KeybindingService]: | Converted keydown event - modifiers: [meta], code: KeyV, keyCode: 52 ('V')
2023-03-01 15:36:14.100 [info] [KeybindingService]: | Resolving meta+[KeyV]
```

**Comment by lramos15** [comment]
Hmmm I wonder what has happening with your keybindings. Here's mine triggering the explorer specific ones with the proper when clauses.
```
2023-03-01 07:51:38.272 [info] [KeybindingService]: | Resolving meta+[KeyC]
2023-03-01 07:51:38.272 [info] [KeybindingService]: \ From 19 keybinding entries, matched filesExplorer.copy, when: explorerViewletVisible && filesExplorerFocus && !explorerResourceIsRoot && !inputFocus, source: built-in.
2023-03-01 07:51:38.273 [info] [KeybindingService]: / Received  keydown event - modifiers: [meta], code: KeyC, keyCode: 67, key: c
2023-03-01 07:51:38.273 [info] [KeybindingService]: | Converted keydown event - modifiers: [meta], code: KeyC, keyCode: 33 ('C')
2023-03-01 07:51:38.274 [info] [KeybindingService]: | Resolving meta+[KeyC]
2023-03-01 07:51:38.274 [info] [KeybindingService]: \ From 19 keybinding entries, matched filesExplorer.copy, when: explorerViewletVisible && filesExplorerFocus && !explorerResourceIsRoot && !inputFocus, source: built-in.
2023-03-01 07:51:38.274 [info] [KeybindingService]: + Invoking command filesExplorer.copy.
2023-03-01 07:51:38.694 [info] [KeybindingService]: / Soft dispatching keyboard event
2023-03-01 07:51:38.694 [info] [KeybindingService]: | Resolving meta+[KeyV]
2023-03-01 07:51:38.695 [info] [KeybindingService]: \ From 4 keybinding entries, matched filesExplorer.paste, when: explorerViewletVisible && filesExplorerFocus && !explorerResourceReadonly && !inputFocus, source: built-in.
2023-03-01 07:51:38.695 [info] [KeybindingService]: / Received  keydown event - modifiers: [meta], code: KeyV, keyCode: 86, key: v
2023-03-01 07:51:38.695 [info] [KeybindingService]: | Converted keydown event - modifiers: [meta], code: KeyV, keyCode: 52 ('V')
2023-03-01 07:51:38.695 [info] [KeybindingService]: | Resolving meta+[KeyV]
2023-03-01 07:51:38.696 [info] [KeybindingService]: \ From 4 keybinding entries, matched filesExplorer.paste, when: explorerViewletVisible && filesExplorerFocus && !explorerResourceReadonly && !inputFocus, source: built-in.
2023-03-01 07:51:38.696 [info] [KeybindingService]: + Invoking command filesExplorer.paste.
2023-03-01 07:51:38.896 [info] [KeybindingService]: + Storing single modifier for possible chord meta.
2023-03-01 07:51:39.196 [info] [KeybindingService]: + Clearing single modifier due to 300ms elapsed.
```

Is your explorer in a non-default position (i.e. not the activity bar)

**Comment by jrieken** [comment]
This gets more and more interesting... The keybinding is there and not (re)configured but it looks like something is wrong with context keys. The `explorerViewletVisible` seems AWOL. Maybe that's because I moved the explorer around very much last endgame?

<img width="2819" alt="Screenshot 2023-03-01 at 16 55 16" src="https://user-images.githubusercontent.com/1794099/222192627-5dc6d216-bbcf-458c-b581-ed09529168e6.png">

**Comment by blakeNaccarato** [comment]
I know this is very early into the next release cycle, but could this be released in a patch to stable VSCode?

I have been bitten by this in stable VSCode after the February 2023 update, by dragging Explorer over to the secondary sidebar. Or is there a workaround to restore functioning of these keys, other than resetting the user profile altogether?

I've tried dragging the Explorer menu back, and also right-clicking and selecting "Reset Location". Neither seems to "restore" the functionality of these keys.

This affects any command using `explorerViewletVisible` in `when`, such as deleting a file with the Delete key. The workaround I managed to do is remove `explorerViewletVisible` from every `when` expression possible, which may have side effects but it works in a pinch. Copy all the keybindings below (expand for details, between the `\\ snip` comments) to fix this in a pinch.

<details><summary>keybindings.json</summary>

```jsonc
[
  // snip
  {
    "key": "shift+delete",
    "command": "deleteFile",
    "when": "filesExplorerFocus && !explorerResourceReadonly && !inputFocus"
  },
  {
    "key": "shift+delete",
    "command": "-deleteFile",
    "when": "explorerViewletVisible && filesExplorerFocus && !explorerResourceReadonly && !inputFocus"
  },
  {
    "key": "delete",
    "command": "deleteFile",
    "when": "filesExplorerFocus && !explorerResourceMoveableToTrash && !explorerResourceReadonly && !inputFocus"
  },
  {
    "key": "delete",
    "command": "-deleteFile",
    "when": "explorerViewletVisible && filesExplorerFocus && !explorerResourceMoveableToTrash && !explorerResourceReadonly && !inputFocus"
  },
  {
    "key": "enter",
    "command": "explorer.openAndPassFocus",
    "when": "filesExplorerFocus && !explorerResourceIsFolder && !inputFocus"
  },
  {
    "key": "enter",
    "command": "-explorer.openAndPassFocus",
    "when": "explorerViewletVisible && filesExplorerFocus && !explorerResourceIsFolder && !inputFocus"
  },
  {
    "key": "ctrl+enter",
    "command": "explorer.openToSide",
    "when": "explorerViewletFocus && !inputFocus"
  },
  {
    "key": "ctrl+enter",
    "command": "-explorer.openToSide",
    "when": "explorerViewletFocus && explorerViewletVisible && !inputFocus"
  },
  {
    "key": "escape",
    "command": "filesExplorer.cancelCut",
    "when": "explorerResourceCut && filesExplorerFocus && !inputFocus"
  },
  {
    "key": "escape",
    "command": "-filesExplorer.cancelCut",
    "when": "explorerResourceCut && explorerViewletVisible && filesExplorerFocus && !inputFocus"
  },
  {
    "key": "ctrl+c",
    "command": "filesExplorer.copy",
    "when": "filesExplorerFocus && !explorerResourceIsRoot && !inputFocus"
  },
  {
    "key": "ctrl+c",
    "command": "-filesExplorer.copy",
    "when": "explorerViewletVisible && filesExplorerFocus && !explorerResourceIsRoot && !inputFocus"
  },
  {
    "key": "ctrl+x",
    "command": "filesExplorer.cut",
    "when": "filesExplorerFocus && !explorerResourceIsRoot && !explorerResourceReadonly && !inputFocus"
  },
  {
    "key": "ctrl+x",
    "command": "-filesExplorer.cut",
    "when": "explorerViewletVisible && filesExplorerFocus && !explorerResourceIsRoot && !explorerResourceReadonly && !inputFocus"
  },
  {
    "key": "shift+alt+f",
    "command": "filesExplorer.findInFolder",
    "when": "explorerResourceIsFolder && filesExplorerFocus && !inputFocus"
  },
  {
    "key": "shift+alt+f",
    "command": "-filesExplorer.findInFolder",
    "when": "explorerResourceIsFolder && explorerViewletVisible && filesExplorerFocus && !inputFocus"
  },
  {
    "key": "space",
    "command": "filesExplorer.openFilePreserveFocus",
    "when": "filesExplorerFocus && !explorerResourceIsFolder && !inputFocus"
  },
  {
    "key": "space",
    "command": "-filesExplorer.openFilePreserveFocus",
    "when": "explorerViewletVisible && filesExplorerFocus && !explorerResourceIsFolder && !inputFocus"
  },
  {
    "key": "ctrl+v",
    "command": "filesExplorer.paste",
    "when": "filesExplorerFocus && !explorerResourceReadonly && !inputFocus"
  },
  {
    "key": "ctrl+v",
    "command": "-filesExplorer.paste",
    "when": "explorerViewletVisible && filesExplorerFocus && !explorerResourceReadonly && !inputFocus"
  },
  {
    "key": "home",
    "command": "firstCompressedFolder",
    "when": "explorerViewletCompressedFocus && filesExplorerFocus && !explorerViewletCompressedFirstFocus && !inputFocus"
  },
  {
    "key": "home",
    "command": "-firstCompressedFolder",
    "when": "explorerViewletCompressedFocus && explorerViewletVisible && filesExplorerFocus && !explorerViewletCompressedFirstFocus && !inputFocus"
  },
  {
    "key": "end",
    "command": "lastCompressedFolder",
    "when": "explorerViewletCompressedFocus && filesExplorerFocus && !explorerViewletCompressedLastFocus && !inputFocus"
  },
  {
    "key": "end",
    "command": "-lastCompressedFolder",
    "when": "explorerViewletCompressedFocus && explorerViewletVisible && filesExplorerFocus && !explorerViewletCompressedLastFocus && !inputFocus"
  },
  {
    "key": "delete",
    "command": "moveFileToTrash",
    "when": "explorerResourceMoveableToTrash && filesExplorerFocus && !explorerResourceReadonly && !inputFocus"
  },
  {
    "key": "delete",
    "command": "-moveFileToTrash",
    "when": "explorerResourceMoveableToTrash && explorerViewletVisible && filesExplorerFocus && !explorerResourceReadonly && !inputFocus"
  },
  {
    "key": "right",
    "command": "nextCompressedFolder",
    "when": "explorerViewletCompressedFocus && filesExplorerFocus && !explorerViewletCompressedLastFocus && !inputFocus"
  },
  {
    "key": "right",
    "command": "-nextCompressedFolder",
    "when": "explorerViewletCompressedFocus && explorerViewletVisible && filesExplorerFocus && !explorerViewletCompressedLastFocus && !inputFocus"
  },
  {
    "key": "left",
    "command": "previousCompressedFolder",
    "when": "explorerViewletCompressedFocus && filesExplorerFocus && !explorerViewletCompressedFirstFocus && !inputFocus"
  },
  {
    "key": "left",
    "command": "-previousCompressedFolder",
    "when": "explorerViewletCompressedFocus && explorerViewletVisible && filesExplorerFocus && !explorerViewletCompressedFirstFocus && !inputFocus"
  },
  {
    "key": "f2",
    "command": "renameFile",
    "when": "filesExplorerFocus && !explorerResourceIsRoot && !explorerResourceReadonly && !inputFocus"
  },
  {
    "key": "f2",
    "command": "-renameFile",
    "when": "explorerViewletVisible && filesExplorerFocus && !explorerResourceIsRoot && !explorerResourceReadonly && !inputFocus"
  }

// snip

]

```

</details>

**Comment by lramos15** [comment]
@blakeNaccarato Does the command in the command palette `Reset View Locations` not fix it for you? This will reset all your views to their default positions.

**Comment by blakeNaccarato** [comment]
My keybindings workaround bypassed the issue on my home computer yesterday. Now I'm at work (both are Windows machines, VSCode up to date), and I went to try your `Reset View Locations` approach. However, I have a lot of menus in my secondary sidebar, so I didn't want to `Reset View Locations` on my main `Default` profile, so I did the following:

- Ensure my `keybindings.json` workaround from before is commented out
- Export the `Default` profile as `Test`
- Import the `Test` profile
- Switch to that profile
- First, try the less invasive, right-click on Explorer button in secondary sidebar and click `Reset Location`
- `Delete` key, and others, works again, even when putting the Explorer back in the secondary sidebar
- Curiously, switching back to the `Default` profile, the `Delete` key, and others, works again, even when putting the Explorer back in the secondary sidebar
- Never had to go the nuclear option and `Reset View Locations`

So, some funny business related to one of the steps above, probably the settings profile shuffle, restored basic functionality of those keys, without having to manually tweak the `when` expressions as before, and without having to go nuclear and `Reset (all) View Locations`.

**TL;DR:** It's possible to work around this with some UI shuffling, but your mileage may vary. My initial request to backport this issue to stable VSCode is not so serious since there are a few workarounds documented here.
